### PR TITLE
[READY]NTnet circuit fix

### LINKED ISSUE #35503: ntnet receivers seem to be broken
bagil 83374
http://take.ms/E2fsb

the "received adress" for some reason its the same of the machines receiving instead of the one sending

https://github.com/tgstation/tgstation/blob/f94f49009ac4fb7d4b758e622ee0638430e5d835/code/modules/integrated_electronics/subtypes/input.dm#L609
something here is borked

### PR DESCRIPTION
**Pull request by arsserpentarium**
[Changelogs]: fixed NTnet circuit.

:cl: NTnet circuit fix
fix: Now NTnet circuits can recieve sender adress properly.Also, now messages could be sended to multiple recepiens.
/:cl:

Note:I've seen system with possibility of multiple recepients. Possibly, Kevinz wanted to do it, but forgot to change one string.

fix #35503

### CODE REVIEW

**Comment by DaxDupont** [comment]
damn i was just working on this

File: code/modules/integrated_electronics/subtypes/input.dm [hunk 1]
```diff
@@ -618,14 +621,15 @@
...
 	var/datum/netdata/data = new
-	data.recipient_ids += target_address
+	data.recipient_ids = splittext(target_address, ";")
+	data.sender_id = address
 	data.plaintext_data = message
 	data.plaintext_data_secondary = text
 	data.plaintext_passkey = key
 	ntnet_send(data)
 
 /obj/item/integrated_circuit/input/ntnet_recieve(datum/netdata/data)
```
Comment (DaxDupont) [inline code/modules/integrated_electronics/subtypes/input.dm]: you might want to move this to 
/obj/item/integrated_circuit/input/ntnet_packet/ntnet_recieve

I tested it and it works

Comment (arsserpentarium) [reply code/modules/integrated_electronics/subtypes/input.dm]: How I will move sending information about sender to recieving proc?

File: code/modules/integrated_electronics/subtypes/input.dm [hunk 2]
```diff
@@ -586,11 +586,12 @@
 	desc = "Enables the sending and receiving of messages on NTNet via packet data protocol."
 	extended_desc = "Data can be send or received using the second pin on each side, \
 	with additonal data reserved for the third pin. When a message is received, the second activation pin \
-	will pulse whatever's connected to it. Pulsing the first activation pin will send a message."
+	will pulse whatever's connected to it. Pulsing the first activation pin will send a message. Message \
+	can be send to multiple recepients. Addresses must be separated with ; symbol."
```
Comment (JStheguy) [inline code/modules/integrated_electronics/subtypes/input.dm]: Just make the target NTNet addresses pin a list pin.

Comment (arsserpentarium) [reply code/modules/integrated_electronics/subtypes/input.dm]: so you should fuck with lists every time you need send message to one adress?
Better to fuck with list concatenator.

Comment (arsserpentarium) [reply code/modules/integrated_electronics/subtypes/input.dm]: By the way.Are you going to reopen your PR with cameras?

**Comment by fludd12** [comment]
You are a gentleman and a scholar.

**Pull request merged**